src/components/Drawer/index.tsx

```diff
@@ -0,0 +1,45 @@
+import { useEffect, useState } from 'react';
+import Portal from '../Portal';
+import { BackDrop, Content, Wrapper } from './styles';
+import { DrawerProps } from './types';
+
+export default function Drawer({
+  type = 'left',
+  open,
+  onClose,
+  children,
+}: DrawerProps) {
+  const [toggle, setToggle] = useState(false);
+
+  useEffect(() => {
+    if (open) {
+      document.body.style.overflow = 'hidden';
+      setToggle((prev) => !prev);
+      return;
+    }
+
+    document.body.removeAttribute('style');
```

Comment:
How about setting `overflow` to `auto` again instead removing whole `style` attribute?